sbt/src/main/scala/pink/cozydev/protosearch/Tasks.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2022 CozyDev
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pink.cozydev.protosearch.sbt
+
+import cats.syntax.all._
+import cats.effect.IO
+import cats.effect.kernel.Resource
+import cats.effect.unsafe.implicits.global
+import sbt.*
+import laika.io.syntax.*
+import laika.io.model.FilePath
+import laika.api.Renderer
+import pink.cozydev.protosearch.analysis.IndexFormat
+import pink.cozydev.protosearch.analysis.Plaintext
+
+object Tasks {
+  import Def.*
+  import laika.sbt.LaikaPlugin.autoImport._
+  import pink.cozydev.protosearch.sbt.ProtosearchPlugin.autoImport._
+
+  val protosearchWriteProtosearchJS: Initialize[Task[Unit]] = task {
+    val targetDir = FilePath.parse(protosearchIndexTarget.value)
+    val filenames = List("protosearch.js", "search.js", "worker.js", "search.html")
+    val path = "pink/cozydev/protosearch/sbt"
+
+    filenames
+      .traverse_ { fn =>
+        val res = IO.blocking(getClass().getClassLoader().getResourceAsStream(s"$path/$fn"))
```

Comment:
just so I don't forget, Arman pointed us to this way of adding an input to the laika virtual file tree https://github.com/typelevel/sbt-typelevel/blob/v0.4.22/site/src/main/scala/org/typelevel/sbt/site/TypelevelHeliumExtensions.scala#L65-L79

We could investigate that in the future

Reply:
It actually now also has `addClassLoaderResource` which would further reduce boilerplate: https://github.com/typelevel/Laika/blob/main/io/src/main/scala/laika/helium/internal/builder/HeliumInputBuilder.scala#L72.

Reply:
It was fixed quite long time ago in 0.19.0 by adding a `ClassLoader` parameter. I would not have left API in 1.x that is known to not work in most circumstances.

Reply:
Yes, we're all fixed up now, but thanks again for the advice :)